d3a_interface/device_validator.py

```diff
@@ -24,10 +25,159 @@
 GeneralSettings = ConstSettings.GeneralSettings
 LoadSettings = ConstSettings.LoadSettings
 PvSettings = ConstSettings.PVSettings
+HomeMeterSettings = ConstSettings.HomeMeterSettings
 StorageSettings = ConstSettings.StorageSettings
 CepSettings = ConstSettings.CommercialProducerSettings
 
 
+class DeviceValidator(metaclass=ABCMeta):
+    """Interface for devices' validator classes."""
+
+    @classmethod
+    @abstractmethod
+    def validate(cls, **kwargs):
```

Comment:
Abstract methods should not define implementations, we should get rid of the `@ abstractmethod` decorator here

Reply:
In theory, we couldn't have a concrete method inside an abstract class, so that would also be a bit unconventional. Python isn't strict about the definition of abstract methods/classes, so it's possible to take advantage of it (see: https://pymotw.com/2/abc/#concrete-methods-in-abcs).

The reason why I made the `validate()` method abstract is that I wanted to make it very explicit that it is necessary for all validators, and it will be called by external objects.
But I definitely see your point and, in terms of final use, it wouldn't make a big difference. I can check again and see if changing it would improve it!